overlay/etc/cron.d/asigra-update.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+
+#TODO
+# - remove the empty lines in INDEX.n
+
+UPDATE="https://asigra-f611.kxcdn.com/14.1"
+FILE=INDEX.prev
+DLDIR=download
+IP=`cat /root/dhcpip`
+
+# Check for if prev index exist if not create
+if [ -f "$FILE" ]; then
+else
+   touch $FILE
+fi
+
+# Check for DLDIR
+if [ -f "$DLDIR" ]; then
+else
+   mkdir $DLDIR
+fi
+
+#fetch index
+curl $UPDATE/INDEX > INDEX
+if [ "$?" -gt 0 ]
+then
+echo "Error downloading INDEX"
+exit 1
+fi
+
+#compare if a new file is there
+diff -ruN INDEX.prev INDEX > INDEX.n
+
+#cut all unneded stuff out
+cut -d: -f4- INDEX.n > INDEX.new
+
+#We need prev. index to compare
+mv INDEX INDEX.prev
+
+#When INDEX.new is empty we dont need to do anything
+if [ ! -s "INDEX.new" ] ; then return 0 ; fi
+
+
+#Get all files from the new index
+#Curl explaintion: we will retry 5 times or a maximum of 40 seconds with a connection timeout of 5 seconds, and no exponential backoff policy
+for i in $(cat INDEX.new)
+do
+	cd $DLDIR && { curl \
+	-O $UPDATE/$i \
+	--fail-early \
+        --connect-timeout 5 \
+        --max-time 300 \
```

Comment:
300 seconds is enough to download 3.6GiB?

Reply:
You are right, thats always dependents on network and location (home or dc), do you think 1200 sec (20 min) would be a good time ?

Reply:
Try 2 hours. 10mbit/s connections are still popular around the world.